crates/spidermonkey-embedding-splicer/src/bindgen.rs

```diff
@@ -516,8 +521,32 @@ impl JsBindgen<'_> {
                             );
                         }
 
-                        if resource.is_some() {
-                            uwriteln!(self.src, "}}");
+                        if let Some(ty) = resource {
+                            let lower_camel = &self.resolve.types[ty]
+                                .name
+                                .as_ref()
+                                .unwrap()
+                                .to_lower_camel_case();
+
+                            let prefix = iface_name
+                                .as_deref()
+                                .map(|s| format!("{s}$"))
+                                .unwrap_or(String::new());
+
+                            let resource_symbol = self.intrinsic(Intrinsic::SymbolResourceHandle);
+                            let dispose_symbol = self.intrinsic(Intrinsic::SymbolDispose);
+
+                            uwriteln!(
+                                self.src,
+                                "
+                                    [{dispose_symbol}]() {{
+                                        finalizationRegistry_import${prefix}{lower_camel}.unregister(this);
+                                        $resource_import${prefix}drop${lower_camel}(this[{resource_symbol}]);
+                                        delete this[{resource_symbol}];
```

Comment:
To nitpick, shape changes in JS have a very slight performance cost as the object gets copied into the new shape. Setting a property to `null` is therefore faster from a microbenchmark perspective, and that would also avoiding the need for the configurability PR.

Reply:
Sounds good.  I'll push an update.